### ISSUE TITLE
Alert icon being shown every time source is switched when Boom 3D is installed

### ISSUE THREAD

**Issue by pedropombeiro** [fredemmott/StreamDeck-AudioSwitcher#47]
**Describe the bug**

When I switch audio source, either by clicking on the action button or by way of a separate multi-action, I see an alert just before the actual source switch goes through.

**To Reproduce**

1. Install Boom 3D audio enhancement software
2. Add a Toggle Audio Device button to the Stream Deck
3. Click on the Toggle Audio Device button
4. Yellow alert icon is shown in button
5. Source is switched

**Expected behavior**

Step 3 should not happen.

**Debugging Information**

I believe this is the line getting hit, when Boom 3D sets the audio output back to itself: https://github.com/fredemmott/StreamDeck-AudioSwitcher/blob/fde9793cb462b26453f85765f3a0d6844e325be4/Sources/AudioSwitcherStreamDeckPlugin.cpp#L141. [Line 251](https://github.com/fredemmott/StreamDeck-AudioSwitcher/blob/master/Sources/AudioSwitcherStreamDeckPlugin.cpp#L251) also gets hit if I switch audio source by other means.

**Log File**

```
18:19:42.0504          void ESDActionsGridViewModel::onKeyPressed(): Key 5 pressed: com.fredemmott.audiooutputswitch.toggle
18:19:42.1513          void ESDActionsGridViewModel::onKeyReleased(): Key 5 released: com.fredemmott.audiooutputswitch.toggle
18:19:45.7356          void ESDActionsGridViewModel::onKeyPressed(): Key 5 pressed: com.fredemmott.audiooutputswitch.toggle
18:19:45.8376          void ESDActionsGridViewModel::onKeyReleased(): Key 5 released: com.fredemmott.audiooutputswitch.toggle
```

** Operating System **

MacOS Catalina

**Additional context**

Using Stream Deck XL and [Boom 3D](https://www.globaldelight.com/boom/)

**Comment by fredemmott** [comment]
> Source is switched

Are you sure this is actually happening, rather than just the streamdeck indicating it?

If so, it's not line 141 - in that case, it returns early on line 142, and never reaches SetDefaultAudioDeviceId

**Comment by fredemmott** [comment]
> when Boom 3D sets the audio output back to itself

What's the exact flow here?

https://github.com/fredemmott/StreamDeck-AudioSwitcher/blob/fde9793cb462b26453f85765f3a0d6844e325be4/Sources/AudioSwitcherStreamDeckPlugin.cpp#L251 seems likely  - an alert will be shown for toggle actions if the current active audio device is *neither* of the configured devices

**Comment by pedropombeiro** [comment]
> seems likely (and desirable) - an alert will be shown for toggle actions if the current active audio device is _neither_ of the configured devices

Yeah, I guess the only workaround here would be to ignore if that audio device happens to be `BoomAudio`, because that is supposed to the the output device at all times when that software is installed on a system:

![2021-06-06 at 21 20](https://user-images.githubusercontent.com/138074/120937295-0fe37800-c70d-11eb-980d-b09e7c8c295b.png)

**Comment by fredemmott** [comment]
Perhaps use the `set device` action in a multi-action switch instead?

The toggle needs to update what it shows if something else changes the device, but for a multi-action switch, that's not expected

**Comment by pedropombeiro** [comment]
> Perhaps use the `set device` action in a multi-action switch instead?

That's what I'm doing in the multi-action button, but the top-level Toggle button shows this alert, which is a bit distracting (and doesn't let me see which device is actually selected). Admittedly, it's a bit of a first-world problem and the plugin works otherwise perfectly...

**Comment by fredemmott** [comment]
Can you grab the demos from https://github.com/fredemmott/AudioDeviceLib/releases/latest, and run the included `list-devices` executable from a terminal?

I think the overall behavior is expected, but if there's a reasonable and predictable way to detect/ignore Boom3d, I'm fine with making it a special case

**Comment by pedropombeiro** [comment]
```
----- INPUT DEVICES -----
"Internal Microphone"
	ID:        "input/AppleHDAEngineInput:1F,3,0,1,0:1"
	Interface: "Built-in Microphone"
	Endpoint:  ""
	State:     CONNECTED
"BoomAudio"
	ID:        "input/GDAudioDevice_UID"
	Interface: "BoomAudio"
	Endpoint:  ""
	State:     CONNECTED
"krisp microphone"
	ID:        "input/KrispAudioDeviceMic"
	Interface: "krisp microphone"
	Endpoint:  ""
	State:     CONNECTED
----- OUTPUT DEVICES -----
"Internal Speakers"
	ID:        "output/AppleHDAEngineOutput:1F,3,0,1,1:0"
	Interface: "Built-in Output"
	Endpoint:  ""
	State:     CONNECTED
"BoomAudio"
	ID:        "output/GDAudioDevice_UID"
	Interface: "BoomAudio"
	Endpoint:  ""
	State:     CONNECTED
"krisp speaker"
	ID:        "output/KrispAudioDeviceSpeaker"
	Interface: "krisp speaker"
	Endpoint:  ""
	State:     CONNECTED
```

**Comment by fredemmott** [comment]
"GDAudioDevice_UID" sounds like a placeholder from an example that was accidentally left in; I'll probably check for that, then fetch the details to check the interface if it matches
